src/main/java/duke/databse/Database.java

```diff
@@ -0,0 +1,215 @@
+            } else {
+                System.out.println(ui.reading_file_message);
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred in creating or opening file.");
+        }
+    }
+
+
+    /**
+     * read data from file
+     * @return
+     */
+    public ArrayList<Task> getData() {
+        ArrayList<Task> objectsList = new ArrayList<>();
+        try {
+            Scanner myReader = new Scanner(file);
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                objectsList.add(parseData(data));
+            }
+        } catch (IOException e) {
+            System.out.println(ui.wrong_message);
+        }
+        return objectsList;
+    }
+
+    /**
+     * parse data from string to task
+     * @param data
+     * @return
+     */
+    public Task parseData(String data) {
+        String[] s = data.split(" ");
+
+        boolean isDone;
+        if (s[s.length - 1].equals("false")) {
+            isDone = false;
+        } else {
+            isDone = true;
+        }
+        switch (s[0]) {
+        case "T":
+            Todo todo = new Todo(s[1], isDone);
+            return todo;
+        case "E":
+            String taskname_event = "";
+            String tasktime_event = "";
+            boolean timepart_event = false;
+            for (int i = 1; i < s.length - 1; i++) {
+                if (s[i].startsWith("/")) {
+                    timepart_event = true;
+                    tasktime_event = s[i].substring(1);
+                } else if (timepart_event) {
+                    tasktime_event += " " + s[i];
+                } else {
+                    if (s[i + 1].startsWith("/")) {
+                        taskname_event += s[i];
+                    } else {
+                        taskname_event += s[i] + " ";
+                    }
+
+                }
+            }
+            Event event = new Event(taskname_event, isDone, tasktime_event);
+            return event;
+        case "D":
+            String taskname_ddl = "";
+            String tasktime_ddl = "";
+            boolean timepart_ddl = false;
+            for (int i = 1; i < s.length - 1; i++) {
+                if (s[i].startsWith("/")) {
+                    timepart_ddl = true;
+                    tasktime_ddl = s[i].substring(1);
+                } else if (timepart_ddl) {
+                    tasktime_ddl += " " + s[i];
+                } else {
+                    if (s[i + 1].startsWith("/")) {
+                        taskname_ddl += s[i];
+                    } else {
+                        taskname_ddl += s[i] + " ";
+                    }
+                }
+            }
+            Deadline deadline = new Deadline(taskname_ddl, isDone, tasktime_ddl);
+            return deadline;
+        default:
+            break;
+        }
+        return null;
+    }
+
+    /**
+     * write data to file
+     * @param todo
```

Comment:
Maybe this header comment should be `Writes data to file.` instead? For the parameter `todo`, we need to add descriptions also.